src/reductive.re

```diff
@@ -92,6 +97,8 @@ module Provider = {
         | Some(unsubscribe) => unsubscribe()
         | None => ()
         },
+      shouldUpdate: ({oldSelf, newSelf}) =>
+        oldSelf.state.reductiveState != newSelf.state.reductiveState,
```

Comment:
As @ambientlight recommended this should really be referential comparison. This will run a quite expensive comparison on deeply nested data structures.

```suggestion
        oldSelf.state.reductiveState !== newSelf.state.reductiveState,
```